akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -71,9 +75,13 @@ private[http2] sealed abstract class Http2ClientDemux
  * In the best case we could just flattenMerge the outgoing side (hoping for the best) but this will probably
  * not work because the sending decision relies on dynamic window size and settings information that will be
  * only available in this stage.
+ *
+ * @param initialRemoteSettings sequence of settings received on the initial header sent from
+ *                              the client in an ' HTTP2-Settings:' header. This parameter should only be used
+ *                              on the server end of a connection.
  */
 @InternalApi
-private[http2] class Http2ServerDemux(http2Settings: Http2CommonSettings, initialDemuxerSettings: immutable.Seq[Setting], upgraded: Boolean, isServer: Boolean) extends GraphStage[BidiShape[Http2SubStream, FrameEvent, FrameEvent, Http2SubStream]] { stage =>
+private[http2] class Http2ServerDemux(http2Settings: Http2CommonSettings, initialRemoteSettings: immutable.Seq[Setting], upgraded: Boolean, isServer: Boolean) extends GraphStage[BidiShape[Http2SubStream, FrameEvent, FrameEvent, Http2SubStream]] { stage =>
```

Comment:
I renamed and documented this param because it confused me over and over.

Reply:
I can see that. TBQH 'remote' can still mean 'the settings we got from the remote' or 'the settings we sent to the remote', I wonder if 'receivedSettings' would be even less ambiguous?